tests/framework/scheduler.py

```diff
@@ -115,96 +241,36 @@ def pytest_runtestloop(self, session):
-                    "/functional/test_rate_limiter.py",
-                    "/functional/test_signals.py",
-                    "/build/test_coverage.py"
-                ],
-                'items': []
-            },
-            {
-                # Safe batch: tests that can be run safely in a concurrent
-                # environment.
-                'name': 'safe',
-                # FIXME: we still have some framework concurrency issues
-                # which prevent us from successfully using `max_concurrency`.
-                # 'concurrency': max_concurrency,
-                'concurrency': 1,
-                'patterns': [
-                    "/functional/.+",
-                    "/build/.+",
-                    "/security/.+"
-                ],
-                'items': []
-            },
-            {
-                # Unknown batch: a catch-all batch, scheduling any tests that
-                # haven't been categorized to run sequentially (since we don't
-                # know if they are concurrency-safe).
-                'name': 'unknown',
-                'concurrency': 1,
-                'patterns': [".+"],
-                'items': []
-            }
-        ]
-
-        # Go through the list of tests and assign each of them to its
-        # corresponding batch in the schedule.
-        for item in session.items:
-            # A test can match any of the patterns defined by the batch,
-            # in order to get assigned to it.
-            for batch in schedule:
-                # Found a matching batch. No need to look any further.
-                if re.search(
-                    "|".join(["({})".format(x) for x in batch['patterns']]),
-                    "/".join(item.listnames()),
-                ) is not None:
-                    batch['items'].append(item)
-                    break
-
-        # Filter out empty batches.
-        schedule = [batch for batch in schedule if batch['items']]
-
-        # Evaluate marker expression only for the marked batch items.
-        # If pytest runs with a marker expression which does not include
-        # `nonci` marked tests (e.g `-m "not nonci" or non-existent marker
-        # expression), the tests marked with `nonci` marker will be skipped.
-        for batch in schedule:
-            PytestScheduler.filter_batch(session.config,
-                                         batch,
-                                         marker_name="nonci")
-            break
-
-        for batch in schedule:
-            self._raw_stdout(
-                "\n[ ",
-                self._colorize('yellow', batch['name']),
-                " | ",
-                "{} tests".format(len(batch['items'])),
-                " | ",
-                "{} worker(s)".format(batch['concurrency']),
-                " ]\n"
-            )
-            self._run_batch(batch)
+        # First, go through the args and check if a file was specified
```

Comment:
what if the user specified a folder? E.G. `tools/devtool test -- integration_tests/style`